### PR TITLE
Refactor Teams Rego Code

### PR DESCRIPTION
**Pull request by Sloane4**
#  #

## 🗣 Description ##

The rego code has been expanded greatly, but: with the addition of Regal (OPA linter), update is style practices, & increase in duplicate code, there is a need for a "clean up".

## 💭 Motivation and context ##

- Not triggering any linter warnings/errors (i.e. line length, some in, identical test names, constant conditions, wildcard & test outside test package)
- Style uniformity with booleans (i.e. `== true`, `== false`, `not ExampleMethod()`,  `ExampleMethod() == true`, and `Method() := true if {}`
- Match util import to file structure `import data.report.utils.NotCheckedDetails` -> `import data.utils.report.NotCheckedDetails`
- Refactor out a general ReportsUtil functions for file to be added to Report Utils file in separate PR
- Simplify code as much as possible by reducing duplicate code
Closing #632

## 🧪 Testing ##

- Unit Tests

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] *All* future TODOs are captured in issues, which are referenced
      in code comments.
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

## ✅ Pre-merge checklist ##

- [ ] Revert dependencies to default branches.
- [ ] Finalize version.

## ✅ Post-merge checklist ##

- [ ] Create a release.

### CODE REVIEW

File: Rego/TeamsConfig.rego [hunk 1]
```diff
@@ -1,277 +1,337 @@
...
-#--
+FilterArray(Conditions, Boolean) := [Condition | some Condition in Conditions; Condition == Boolean]
+
+#############
+# Constants #
+#############
+
+FAIL := ReportDetailsBoolean(false)
+
+PASS := ReportDetailsBoolean(true)
```
[the remark this answers is not in the document]
Comment (Sloane4) [reply Rego/TeamsConfig.rego:20]: In PR #642 , all duplicate constants & functions are going to be moved to the appropriate utils function. I can make sure if there is anything left over in the file that constants go before methods as a basic style rule.

File: Rego/TeamsConfig.rego [hunk 2]
```diff
@@ -281,293 +341,357 @@ tests[{
...
-	# According to Get-CsTeamsClientConfiguration, is team email integration enabled?
+# Concat the AssignedPlan for each tenant in one comma seperated string
+AssignedPlans := concat(", ", TenantConfig.AssignedPlan) if {
+    some TenantConfig in input.teams_tenant_info
+}
+
+# If AssignedPlan (one of the tenant configs) contain the string
+# "GCC" and/or "DOD", return true, else return false
+default TenantType := false
+TenantType := true if {
```
[the remark this answers is not in the document]
Comment (Sloane4) [reply Rego/TeamsConfig.rego]: updated, please confirm & resolve

**Comment by nanda-katikaneni** [comment]
@Sloane4 Run-OPA Lint tests seems to be failing - can you take a look.

**Comment by Sloane4** [comment]
@nanda-katikaneni fixed

**Pull request merged**